src/components/CountrySelector/index.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+import * as React from 'react'
+import { h, Component } from 'preact'
+import classNames from 'classnames'
+import PageTitle from '../PageTitle'
+import Button from '../Button'
+import { localised } from '../../locales'
+import type { LocalisedType } from '../../locales'
+import { getSupportedCountriesForDocument } from '../../supported-documents'
+import { trackComponent } from 'Tracker'
+import { preventDefaultOnClick } from '~utils/index'
+import theme from 'components/Theme/style.scss'
+import style from './style.scss'
+
+import Autocomplete from 'accessible-autocomplete/preact'
+import 'accessible-autocomplete/dist/accessible-autocomplete.min.css'
+
+type Props = {
+  documentType: string,
+  previousStep: () => void,
+  nextStep: () => void,
+  translate: (string, ?{}) => string,
+  actions: Object,
+} & LocalisedType
+
+type State = {
+  hasError: Boolean,
+}
+
+class CountrySelection extends Component<Props, State> {
+  state = {
+    hasError: false,
+  }
+
+  handleCountrySelect = (selectedCountry: Object) => {
+    if (selectedCountry) {
+      this.setState({
+        hasError: false,
+      })
+      this.props.actions.setIdDocumentIssuingCountry(selectedCountry)
+    } else {
+      this.setState({
+        hasError: true,
+      })
+    }
```

Comment:
from my understanding this function is when a user selects a country from the dropdown right? if my understanding is correct then what is the error state here?

Reply:
It's quite subtle, but there is a change in text and the icon transitions from a blue ❓ icon to a red ❗ icon if there is no country selected.
The country selection dropdown itself has its own "no results" error state. But this acts an additional warning to the user why the button is still disabled. I probably should give it a better naming. 🤔

Reply:
yeah from the function name I assumed at this state a country was selected. But this also accounts for the country being typed with country not being set ("no results" right?)

Reply:
Yeap, that's right.